src/test/java/io/redis/test/utils/EnabledOnCommandRule.java

```diff
@@ -0,0 +1,126 @@
+package io.redis.test.utils;
+
+import io.redis.test.annotations.EnabledOnCommand;
+import org.junit.Assume;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import redis.clients.jedis.*;
+import redis.clients.jedis.resps.CommandInfo;
+
+import java.lang.reflect.Method;
+import java.util.Map;
+
+
+public class EnabledOnCommandRule implements TestRule {
+    private static final Logger logger = LoggerFactory.getLogger(EnabledOnCommandRule.class);
+
+    private final HostAndPort hostPort;
+    private final JedisClientConfig config;
+
+    public EnabledOnCommandRule(HostAndPort hostPort, JedisClientConfig config) {
+        this.hostPort = hostPort;
+        this.config = config;
+    }
+
+    public EnabledOnCommandRule(EndpointConfig endpointConfig) {
+        this.hostPort = endpointConfig.getHostAndPort();
+        this.config = endpointConfig.getClientConfigBuilder().build();
+    }
+
+    @Override
+    public Statement apply(Statement base, Description description) {
+        return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                try (Jedis jedisClient = new Jedis(hostPort, config)) {
+                    String[] command = getCommandFromAnnotations(description);
+
+                    if (command != null && !isCommandAvailable(jedisClient, command[0],command[1])) {
```

Comment:
I am afraid that it will add additional complexity to test infra and a chance for erroneous behavior. I wouldn't go with such optimization unless we see it is required.,